osquery/filesystem/windows/fileops.cpp

```diff
@@ -1192,13 +1192,26 @@ boost::optional<FILE*> platformFopen(const std::string& filename,
   return fp;
 }
 
-Status namedPipeExists(const std::string& path) {
-  // Wait 500ms for the named pipe status
-  if (::WaitNamedPipeA(path.c_str(), 500) == 0) {
+/**
+ * @brief The windows implementation introduces a 500ms max wait.
+ *
+ * We cannot use existing methods to determine the lifespan of the
+ * extensions/extensions manager socket. On Windows, the Thrift install is
+ * brittle and does not like a quick connect and disconnect. To compensate, we
+ * use WaitNamedPipe to determine the existence of a named pipe. If the named
+ * pipe does not exist, WaitNamedPipe should error with ERROR_BAD_PATHNAME.
+ */
+Status socketExists(const fs::path& path, bool remove_socket) {
```

Comment:
This logic is mutiplexed with:
```cpp
if ((::WaitNamedPipeA(path.c_str(), NAMED_PIPE_WAIT) == 0) &&		
    (::GetLastError() == ERROR_BAD_PATHNAME)) {		
  return Status(1, "Named pipe path is invalid");		
}
```

Would the pipe path be mangled when converted into an `fs::path`?